src/core/include/lattice/hal/default/dcrtpoly-impl.h

```diff
@@ -1926,128 +1926,105 @@ DCRTPolyImpl<VecType> DCRTPolyImpl<VecType>::ScaleAndRound(
     }
 
     std::vector<NativeInteger> mu(sizeO);
-    for (usint j = 0; j < sizeO; j++) {
+    for (size_t j = 0; j < sizeO; ++j)
         mu[j] = (paramsOutput->GetParams()[j]->GetModulus()).ComputeMu();
-    }
 
 #if defined(HAVE_INT128) && NATIVEINT == 64
     #pragma omp parallel for
-    for (usint ri = 0; ri < ringDim; ri++) {
+    for (usint ri = 0; ri < ringDim; ++ri) {
         double nu = 0.5;
-
-        for (usint i = 0; i < sizeI; i++) {
+        for (size_t i = 0; i < sizeI; ++i) {
+            // possible loss of precision if modulus greater than 2^53 + 1
             const NativeInteger& xi = m_vectors[i + inputIndex][ri];
-            nu += tOSHatInvModsDivsFrac[i] * xi.ConvertToInt();
+            nu += tOSHatInvModsDivsFrac[i] * xi.ConvertToDouble();
         }
-
         if (!is64BitOverflow(nu)) {
-            NativeInteger alpha = static_cast<uint64_t>(nu);
-
-            for (usint j = 0; j < sizeO; j++) {
-                DoubleNativeInt curValue = 0;
-
-                const NativeInteger& oj                                  = paramsOutput->GetParams()[j]->GetModulus();
-                const std::vector<NativeInteger>& tOSHatInvModsDivsModoj = tOSHatInvModsDivsModo[j];
-
-                for (usint i = 0; i < sizeI; i++) {
+            NativeInteger alpha = static_cast<BasicInteger>(nu);
```

Comment:
Change BasicInteger to uint64_t

Reply:
Will do. Just note that
`NativeInteger alpha = static_cast<uint64_t>(nu);`
becomes
`NativeInteger alpha = static_cast<BasicInteger>(static_cast<uint64_t>(nu));`
if BasicInteger != uint64_t

Reply:
which could happen at L1995
https://github.com/openfheorg/openfhe-development/blob/de24e7e423284ca03c9d8c7305bec5f591b60bc5/src/core/include/lattice/hal/default/dcrtpoly-impl.h#LL1995C16-L1995C16

Reply:
it would be an issue only for NATIVE_SIZE = 32. Currently NATIVE_SIZE = 32 should only be used for binfhe. So probably the change I suggested is not needed.